packages/grafana-data/src/field/displayProcessor.ts

```diff
@@ -137,7 +142,7 @@ export function getDisplayProcessor(options?: DisplayProcessorOptions): DisplayP
       percent = scaleResult.percent;
     }
 
-    return { text, numeric, prefix, suffix, color, percent };
+    return { text, numeric, prefix, suffix, color, icon, percent };
```

Comment:
Maybe:

```suggestion
    const disp = { text, numeric, prefix, suffix, color };
    if (icon != null) {
       disp.icon = icon;
    }
    if (percent != null) {
       disp.percent = percent;
    }
    return disp;
```

would be nicer not to have all the explicit undefined values defined in each object.